### PR TITLE
Filter taxids out of hit summaries

### PR DESCRIPTION
**Pull request by morsecodist**
This actually contains two changes, a bit confusing but one won't work without the other.

First, I make taxids all strings. We have a big problem in the codebase of taxids being strings/numbers. For filtering, they must be strings. In some places they had to be ints because we do math with them and compare them as numbers sometimes. I think a cleaner approach overall would actually be to make them always ints but it is much safer to go in the string direction since this is explicitly used in way more places and we very rarely do math on the ids.

Second, I filter out taxids from the hit summary from alignment. This has the effect of filtering them out of the refined hits ummary as well. It took a lot of analyzing the code to come to this conclusion but I am pretty confident about it at this point. I added an explainatory comment. In general no taxids will be in the refined hit summary if they are not in the hit summary.

### CODE REVIEW

**Review by tfrcarvalho — COMMENTED** [review]
My memory of the original issue is a bit fuzzy, but it does look like you address the original issue of filtering out of `output_hitsummary`.
A couple of comments:
* I understand but am not fond of making taxIds into string instead of guaranteeing they are ints. Are we in any way exposed to bad taxids? Do we need to do any validation before accepting input as strings?
* How did you test it? Did you go over the edge cases in the ticket?

In general,  it would be good to include a testing section describing how the solution was tested.
nit: When there is a mapping to a clubhouse ticket, it would be great to include it in the title and/or description.

File: short-read-mngs/idseq-dag/idseq_dag/util/lineage.py [hunk 1]
```diff
@@ -58,33 +54,33 @@ def fill_missing_calls(cleaned_lineage: List[str]) -> List[str]:
...
-        tax_level -= 1
     return result
 
 
-def blank(taxid_int):
+def _blank(taxid_int: int):
     return 0 > taxid_int > INVALID_CALL_BASE_ID
 
 
 def fill_missing_calls_tests():
```
Comment (tfrcarvalho) [inline short-read-mngs/idseq-dag/idseq_dag/util/lineage.py:70]: Do you know the history behind this? Is it still being used? Could not find it

Comment (morsecodist) [reply short-read-mngs/idseq-dag/idseq_dag/util/lineage.py:70]: I did a global search and I could not. I made as many of the functions in this module private as I could to avoid confusion.

**Comment by morsecodist** [comment]
@tfrcarvalho 

> Are we in any way exposed to bad taxids?

They are strings in the lookup dicts but they shouldn't be bad anywhere else. I agree the strings are a bit of a hack. I'd prefer to go ints but I felt it was a bit riskier. Do you think I should go ints?

Testing wrietup incoming

File: short-read-mngs/idseq-dag/idseq_dag/util/lineage.py (hunk 1 above)
Comment (tfrcarvalho) [reply short-read-mngs/idseq-dag/idseq_dag/util/lineage.py:70]: Not worth deleting?

**Comment by tfrcarvalho** [comment]
> They are strings in the lookup dicts but they shouldn't be bad anywhere else. I agree the strings are a bit of a hack. I'd prefer to go ints but I felt it was a bit riskier. Do you think I should go ints?

You have a better understanding of the immediate effort/risk (if our inputs do not have the chance of having malformed tax ids or the parser check the format , there is not much of a risk)... so, will leave final decision up to you.
